mycore-base/src/main/java/org/mycore/crypt/MCRCryptCommands.java

```diff
@@ -0,0 +1,160 @@
+            cipher.init(cipherid);
+            LOGGER.info("Keyfile generated.");
+        } catch (MCRCryptCipherConfigurationException e) {
+            LOGGER.info(e.getStackTraceAsString());
+            LOGGER.info("Error while initialising cipher. No keyfile generated. ");
+        } catch (FileAlreadyExistsException e) {
+            LOGGER.info("Keyfile already exists. No keyfile generated.");
+        }
+    }
+    
+    /**
+     * generate and overwrite of keyfile of cipher {0} 
+     * 
+     * @param cipherid
+     *            String id of cipher configured in properties
+     */
+    @MCRCommand(syntax = "overwrite keyfile for cipher {0}",
+        help = "The command generate on overwrite the keyfile for the cipher configured in mycore.properties."
+               + " Fails if key file exists.",
+        order = 20)
+    public static void overwriteKeyFile(String cipherid) {
+        LOGGER.info("Start genrerateKeyFile");
+        try {
+            MCRCipher cipher = MCRCipherManager.getUnIntitialisedCipher(cipherid);
+            cipher.overwriteKeyFile();
+            cipher.init(cipherid);
+            LOGGER.info("Keyfile overwriten.");
+        } catch (MCRCryptCipherConfigurationException e) {
+            LOGGER.info(e.getStackTraceAsString());
+            LOGGER.info("Error while initialising cipher. No keyfile generated. ");
+        } 
+        
+    }
+    
+    /**
+     * encrypt {0} with cipher {1} 
+     * 
+     * @param value
+     *            The value to be encrypted
+     * 
+     * @param cipherid
+     *            String id of cipher configured in properties
+     */
+    @MCRCommand(syntax = "encrypt {0} with cipher {1}",
+        help = "The command encrypt the value with cipher.",
+        order = 30)
+    public static void encrypt(String value, String cipherid) {
+        try {
+            MCRCipher cipher = MCRCipherManager.getCipher(cipherid);
+            LOGGER.info("Encrypted Value: " + cipher.encrypt(value));
+        } catch (MCRCryptKeyNoPermission e) {
+            LOGGER.info("Value not encrypted. No Permission for chipher " + cipherid + ".");
+        } catch ( MCRCryptCipherConfigurationException e) {
+        	LOGGER.info(e.getStackTraceAsString());
+        	LOGGER.info("Value not encrypted. Invalid configuration or key.");
+        }
+    }
+    
+    /**
+     * decrypt {0} with cipher {1} 
+     * 
+     * @param value
+     *            The value to be decrypted
+     * 
+     * @param cipherid
+     *            String id of cipher configured in properties
+     */
+    @MCRCommand(syntax = "decrypt {0} with cipher {1}",
+        help = "The command encrypt the value with cipher.",
+        order = 40)
+    public static void decrypt(String value, String cipherid) {
+        try {
+            MCRCipher cipher = MCRCipherManager.getCipher(cipherid);
+            LOGGER.info("Decrypted Value: " + cipher.decrypt(value));
+        } catch (MCRCryptKeyNoPermission e) {
```

Comment:
I think it is better to just throw the Exception, so the command will end with a error, if it can not decrypt the key.

Reply:
will changed